app/assets/javascripts/request_viewer_state.js

```diff
@@ -0,0 +1,88 @@
+/* eslint-disable camelcase */
+/* global Blacklight */
+
+(function (global) {
+  const RequestViewerState = {
+      init: function() {
+        this.setupIframeMessageListener();
+        this.setupRequestButton();
+        $('[data-embed-url]').oEmbed();
+      },
+
+      setupRequestButton: function() {
+        document.querySelector('#request-state').addEventListener('click', (event) => {
+          this.formId = event.target.dataset.formId;
+          this.itemId = event.target.dataset.itemId;
+          this.requestState();
+        });
+      },
+
+      setupIframeMessageListener: function() {
+          window.addEventListener('message', (event) => {
+              if (event && event.data) {
+                // avoid development issues
+                if (event.data == "recaptcha-setup" || event.data.source == "react-devtools-content-script") { return; }
+                let parsedData;
+                try {
+                    parsedData = typeof event.data === 'string' ? JSON.parse(event.data) : event.data;
+                } catch (error) {
+                    console.error('Failed to parse event data:', error);
+                    return; // Exit if parsing fails
+                }
+
+                if (parsedData.type === "stateResponse" && parsedData.source === "sul-embed-m3") {
+
+                    console.log('Exhibits: received state:', parsedData.data);
```

Comment:
Please extract this block to a separate method. The outer method is very long and hard to fully understand.